### PR TITLE
Add support for bot config via Zulip, improve workflows across the board

### PR DESCRIPTION
**Pull request by QEDK**
Lots of changes in this one:
* Goodbye `if-else` , welcome `dict()` 👋 
* Improved parsebot workflow by removing `projects.json` from version control in some cases.
* De-duplicated some `goodbot` code 🔢 🔢 
* Updated a few old instructions 🆕 
* Modularized all `goodbot` components that will need to be updated every season 📦 
* Re-wrote ircbot components for reliability and uptime 🆙 
* Add support for bot configuration via Zulip 🤖 ⚡

### CODE REVIEW

File: bots/goodbot.py [hunk 1]
```diff
@@ -278,17 +279,69 @@ def process(self, msg):
...
 					"to": destination,
 					"topic": topic,
 					"content": f"{greeting} :blush: What can I do for you today?"
 				})
-			else:
-				return
-		except Exception as e:
-			print(e)
+
+			keywords = {
```
Comment (xSavitar) [inline bots/goodbot.py:336]: Nice!

Comment (xSavitar) [reply bots/goodbot.py:336]: To make this more readable and easy to add if things become plenty, put each key word and value pair on it's own line and use the almighty sorting to keep it clean 👍

Comment (QEDK) [reply bots/goodbot.py:336]: Fixed in 3a900185233bed1bd2210328232ef1481be60557

**Comment by QEDK** [comment]
@xSavitar Why not! I've wrapped up your suggestion and added authentication as well now, so I think it's pretty much done! :)

**Comment by QEDK** [comment]
@xSavitar Why not! I've wrapped up your suggestion and added authentication as well now, so I think it's pretty much done! :)

**Pull request merged**